### PR TITLE
Chore/de 4264 trx align cfx with upstream pooja

### PR DESCRIPTION
**Pull request by poojapatel23**

### CODE REVIEW

**Review by viannbreslerAccso — APPROVED** [review]

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/AssetAsBuiltControllerAuthorizationIT.java [hunk 1]
```diff
@@ -46,12 +46,11 @@ void shouldAllowAsBuiltEndpointOnlyForSpecificRoles(JwtRole role, boolean isAllo
         given()
                 .header(oAuth2Support.jwtAuthorizationWithOptionalRole(role))
                 .contentType(ContentType.JSON)
-        .when()
+                .when()
```
Comment (rogocof) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/AssetAsBuiltControllerAuthorizationIT.java:49]: do we have a specific formatig config file for IntelliJ? was the formatting just wrong on the upstream side or do we have non harmonized format settings in our team?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java [hunk 2]
```diff
@@ -37,7 +37,14 @@ class DashboardControllerAuthorizationIT extends IntegrationTestSpecification {
 
     private static final String ROOT = "/api/dashboard";
 
-    @ParameterizedTest
+    /*
+     * TODO (Pooja):
```
Comment (rogocof) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java:41]: general question: Is it allowed to leave "TODO" comments in PRs. I understand the necessity here but still want to clarify if it is allowed or not

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/bpn/mapping/BpnMappingControllerAuthorizationIT.java [hunk 3]
```diff
@@ -43,65 +43,89 @@ class BpnMappingControllerAuthorizationIT extends IntegrationTestSpecification {
...
-    @ParameterizedTest
+    /*
+     * TODO (Pooja):
+     * - Task DO-5327
+     * - Review and update PreAuthorize condition in BPNMapping Controller as per Cofinity requirement
+     * - According to Cofinity requirement, only ADMIN has permission
+     * - According to Upstream requirement, ADMIN and SUPERVISOR have permission
+     */
+    // @ParameterizedTest
     @MethodSource("org.eclipse.tractusx.traceability.integration.common.support.RoleSupport#adminRoleAllowed")
```
Comment (rogocof) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/bpn/mapping/BpnMappingControllerAuthorizationIT.java:54]: as we use the String parameter here quite often, wouldn't it make sense to introduce a constant here to not write it every time?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/alert/ReadCreatedAlertsInSortedOrderControllerIT.java [hunk 4]
```diff
@@ -101,69 +106,68 @@ void givenSortByDescriptionProvided_whenGetAlerts_thenReturnAlertsProperlySorted
...
+                .body("content.severity", Matchers.containsInRelativeOrder("CRITICAL", "LIFE-THREATENING", "MAJOR", "MINOR"));
     }
 
     @Test
     void givenInvalidSort_whenGetCreated_thenBadRequest() throws JoseException {
         // given
         String sortString = "createdDate,failure";
+        String filterString = "channel,EQUAL,SENDER,AND";
 
-        // when/then
```
Comment (rogocof) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/alert/ReadCreatedAlertsInSortedOrderControllerIT.java:157]: can we harmonize the // given // when // then commenting here? sometimes it is ommited, sometimes caps sometimes all in one comment

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/InvestigationControllerFilterIT.java [hunk 5]
```diff
@@ -364,10 +364,11 @@ void givenInvestigations_whenGetInvestigationsByAssetId_thenReturnOnlyRelatedInv
         // Given
         assetsSupport.defaultAssetsStored();
         AssetAsBuiltEntity assetAsBuilt = jpaAssetAsBuiltRepository.findById("urn:uuid:d387fa8e-603c-42bd-98c3-4d87fef8d2bb").orElseThrow();
-        AssetAsBuiltEntity assetAsBuilt2 = jpaAssetAsBuiltRepository.findById("urn:uuid:7fa65f10-9dc1-49fe-818a-09c7313a4562").orElseThrow();
         investigationsSupport.storeInvestigationWithStatusAndAssets(CREATED, List.of(assetAsBuilt), null);
         investigationsSupport.storeInvestigationWithStatusAndAssets(SENT, List.of(assetAsBuilt), null);
         investigationsSupport.storeInvestigationWithStatusAndAssets(RECEIVED, List.of(assetAsBuilt), null);
+
+        AssetAsBuiltEntity assetAsBuilt2 = jpaAssetAsBuiltRepository.findById("urn:uuid:7fa65f10-9dc1-49fe-818a-09c7313a4562").orElseThrow();
```
Comment (rogocof) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/InvestigationControllerFilterIT.java:371]: is it possible to be more descriptive in the variable naming? I personally do not understand the difference between assetAsBuilt and assetAsBuilt2

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/ReadReceivedInvestigationsSortedWithSearchCriteriaIT.java [hunk 6]
```diff
@@ -76,54 +77,46 @@ private static Stream<Arguments> sortAndFilterArguments() {
     @ParameterizedTest
     @MethodSource("sortAndFilterArguments")
     void givenSortAndTwoStatusFilters_whenCallSortAndFilterEndpoint_thenReturnExpectedResponse(
-            final long page,
-            final long size,
+            final int page,
+            final int size,
             final String sort,
             final String filter1,
```
Comment (rogocof) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/ReadReceivedInvestigationsSortedWithSearchCriteriaIT.java:83]: general question: Why do we have these numerations in our variable naming? Is their some semantic system behind?

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java [hunk 7]
```diff
@@ -37,7 +37,14 @@ class DashboardControllerAuthorizationIT extends IntegrationTestSpecification {
...
+     * - Task DO-5327
+     * - Add PreAuthorize condition in Dashboard Controller as per Cofinity requirement
+     * - According to Cofinity requirement, ADMIN, SUPERVISOR, USER have permission
+     * - According to Upstream requirement, there should be no PreAuthorize
+     */
+    //@ParameterizedTest
```
Comment (rogocof) [inline tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java:47]: if we comment out that line, we should maybe also comment out the import in line 26 as it is unused

**Review by mauantoniojw — APPROVED** [review]

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java (hunk 2 above)
Comment (poojapatel23) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java:41]: Yes It's allowed to leave "TODO" comment specially for this merge task. My next task after merge,  will be fix TODO task after discuss with people.

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java (hunk 7 above)
Comment (poojapatel23) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/DashboardControllerAuthorizationIT.java:47]: Removed  :)

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/alert/ReadCreatedAlertsInSortedOrderControllerIT.java (hunk 4 above)
Comment (poojapatel23) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/alert/ReadCreatedAlertsInSortedOrderControllerIT.java:157]: try to commented in everywhere :)

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/bpn/mapping/BpnMappingControllerAuthorizationIT.java (hunk 3 above)
Comment (poojapatel23) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/bpn/mapping/BpnMappingControllerAuthorizationIT.java:54]: I don't know why but we can check after stable environment(after merge with upstream)

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/AssetAsBuiltControllerAuthorizationIT.java (hunk 1 above)
Comment (poojapatel23) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/assets/AssetAsBuiltControllerAuthorizationIT.java:49]: I dont think so but generally I and Heinrich use Intellij Reformate Code function

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/ReadReceivedInvestigationsSortedWithSearchCriteriaIT.java (hunk 6 above)
Comment (poojapatel23) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/ReadReceivedInvestigationsSortedWithSearchCriteriaIT.java:83]: I just followed old code ;(

File: tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/InvestigationControllerFilterIT.java (hunk 5 above)
Comment (poojapatel23) [reply tx-backend/src/test/java/org/eclipse/tractusx/traceability/integration/qualitynotification/investigation/InvestigationControllerFilterIT.java:371]: I can't change here but definitly after merge we will think about it

**Review by rogocof — APPROVED** [review]
thanks for answering my comments. From my perspective, we can merge

**Pull request merged**